addons/payment/models/payment_acquirer.py

```diff
@@ -336,6 +336,13 @@ def write(self, vals):
         self._check_required_if_provider()
         return result
 
+    def unlink(self):
+        """ prevent the deletion of the payment acquirer if the corresponding module is available. """
+        for acquirer in self:
+            if acquirer.module_id:
+                raise UserError(_('You cannot delete this payment acquirer because it has a correspondence model.'))
```

Comment:
I'd prefer more general solution and check if the record has xmlid or not.
I.e. similar to https://github.com/odoo/enterprise/commit/39d34f928c147e31a4157a2eb6283c7114de7a4c

v14 doesn't have `@api.ondelete`, so I'd target v16 instead

Reply:
Yes, you can. I just feel that v15 is more stable than v16, so I'd avoid unnecessary changes in v15